File: 1-js/1-js-basics/7.1-more-exercises/2-conditional-statements/01-pipesInPool.js
function solve([vL, debitOneH, debitTwoH, h]) {

    vL = Number(vL);

    debitOneH = Number(debitOneH);

    debitTwoH = Number(debitTwoH);

    h = Number(h);

    let debitOneTotal = debitOneH * h;

    let debitTwoTotal = debitTwoH * h;

    let totalDebit = debitOneTotal + debitTwoTotal;

    return totalDebit <= vL ? `The pool is ${((totalDebit / vL) * 100).toFixed(2)}% full. Pipe 1: ${((debitOneTotal / totalDebit) * 100).toFixed(2)}%. Pipe 2: ${((debitTwoTotal / totalDebit) * 100).toFixed(2)}%.` : `For ${h.toFixed(2)} hours the pool overflows with ${(totalDebit - vL).toFixed(2)} liters.`;

}

console.log(solve(['1000', '100', '120', '3']));
console.log(solve(['100', '100', '100', '2.5']));

File: 2-python/1-python-basics/2.3-conditional-statements-more-exercise/01-pipes_in_pool.py
pool_v = int(input())
first_pipe_debit_h = int(input())
second_pipe_debit_h = int(input())
worker_absence_time_h = float(input())

total_debit_first_pipe = first_pipe_debit_h * worker_absence_time_h
total_debit_second_pipe = second_pipe_debit_h * worker_absence_time_h
total_debit_both_pipes = total_debit_first_pipe + total_debit_second_pipe
output = ''

if total_debit_both_pipes <= pool_v:
    curr_pool_state = total_debit_both_pipes / pool_v * 100
    total_debit_first_pipe_percent = total_debit_first_pipe / total_debit_both_pipes * 100
    total_debit_second_pipe_percent = total_debit_second_pipe / total_debit_both_pipes * 100
    output = f'The pool is {curr_pool_state:.2f}% full.' \
             f'Pipe 1: {total_debit_first_pipe_percent:.2f}%. ' \
             f'Pipe 2: {total_debit_second_pipe_percent:.2f}%.'
else:
    l_diff = total_debit_both_pipes - pool_v
    output = f'For {worker_absence_time_h:.2f} hours the pool overflows with {l_diff:.2f} liters.'

print(output)
